### ISSUE TITLE
Exception when lock file has illegal file content

### ISSUE THREAD

**Issue by vatbub** [prat-man/unique4j#2]
Thanks for the great library. I am using it for a project of mine and found a bug while using it. I am already working on a fix and will submit the pull request later today. I just wanted to document the issue and tell you what's going on. Cheers :)

# Current situation

For an unknown reason, I ended up in a state where the lock file was still in my temp folder, but it was empty. Hence, when trying to acquire a new lock, the following exception occurred:

```
java.lang.NumberFormatException: null
        at java.lang.Integer.parseInt(Unknown Source)
        at java.lang.Integer.parseInt(Unknown Source)
        at tk.pratanumandal.unique4j.Unique.lockFile(Unique.java:352)
        at tk.pratanumandal.unique4j.Unique.acquireLock(Unique.java:173)
        at com.github.vatbub.smartcharge.EntryClass$Companion.main(EntryClass.kt:62)
        at com.github.vatbub.smartcharge.EntryClass.main(EntryClass.kt)
        ... 11 more
```

# Cause
The method `lockFile()` is called to determine the port number of another `unique4j` instance. This method tries to read the contents of the lock file and parse them as an integer. In my case, the file was empty, causing `Integer.parseInt(...)` to throw said `NumberFormatException`.

# Desired behaviour
- The exception shall be caught. 
- The application shall assume that there is no other instance running at the moment.
- The faulty lock file hall be deleted so that it can be replaced by a good one.

**Comment by prat-man** [comment]
I am really honoured by your interest in my library. I am glad you find it useful.
It is always very rewarding to see one's work used and appreciated.

I thank you for pointing out the issue. I shall merge your code, comment it, and publish it in the next release as soon as possible. Thank you for taking the time to improve my library. It means a lot and will surely help other users as well.

I shall let you know once I have published the next release with your fix.

**Comment by prat-man** [comment]
I have pulled your fix and found an even simpler solution and have pushed it to master which is to become version 1.2.2. If you wish, you can check if the fixed code is working for you.

I shall let you know when version 1.2.2 is released.

**Comment by vatbub** [comment]
I wasn't sure whether I could simply leave the corrupted file on disk or not when the exception occurs, hence, I deleted it, but the fix looks great. Thanks :)

**Comment by prat-man** [comment]
I have released Unique4j version 1.3. It comes with a multiple bug fixes (including this bug fix), and new functionality. I highly recommend that you upgrade to version 1.3 to fix that issue.

**Comment by vatbub** [comment]
Thanks for the info, I just updated to version 1.3 :)
